janus.c

```diff
@@ -3680,6 +3667,7 @@ gint main(int argc, char *argv[])
 		gboolean private_address = FALSE;
 		const char *test_ip = nat_1_1_mapping ? nat_1_1_mapping : local_ip;
 		struct sockaddr_in addr;
+		/* FIXME This ugly check only works with IPv4... */
```

Comment:
What you need here is a helper function in `ip-utils` like:

```
// addr_type may be one of *_ipv4, *_ipv6 or *_any_ip
// user_value is the string to convert
// result will be populated if successful.
// returns 0 on success, -EINVAL otherwise (if any argument is invalid).
int janus_network_string_to_address(janus_ip_network_query_options addr_type, const char *user_value, janus_network_address *result);

// convenience version of the above function prototype, passes *_any_ip to the full version.
int janus_network_string_to_address_default(const char *user_value, janus_network_address *result);
```

The guts for that function could be copied from `janus_prepare_network_device_query` and adapted as needed.